### DISCUSSION TITLE
How to update software correctly?

### DISCUSSION THREAD

**Discussion by MikleBas** [magicbug/Cloudlog#1402] [Q&A]
How to update software correctly?
I get errors when I type git pull command (see below)
Also, the name of the executable script for updating is not correctly given in your cron schedule
The advice says
0 0 * * * /bin/bash -c "<Full-Path-To-Bash-Script>/cloudlog.sh"
but in fact there is no such script, but there is one with the name update_cloudlog.sh*
```
hint: Pulling without specifying how to reconcile divergent branches is
hint: discouraged. You can squelch this message by running one of the following
hint: commands sometime before your next pull:
clue:
hint: git config pull.rebase false # merge (the default strategy)
hint: git config pull.rebase true # rebase
hint: git config pull.ff only # fast-forward only
clue:
hint: You can replace "git config" with "git config --global" to set a default
hint: preference for all repositories. You can also pass --rebase, --no-rebase,
hint: or --ff-only on the command line to override the configured default per
hint: invocation.
Update cb7ae557..7a3c80b0
error: Your local changes to the specified files will be overwritten when merging:
        application/libraries/AdifHelper.php
        application/views/adif/data/clublog.php
        application/views/adif/data/exportall.php
        application/views/adif/data/exportsat.php
        application/views/backup/exportall.php
Commit or stash your changes before merging branches.
interrupt
```

**Comment by ea5wa** [comment]
Same question here.... how is the best way to update Cloudlog?

**Comment by ea5wa** [comment]
Ok. I FOUND IT....Running ./update_cloudlog.sh from Git Bash 

![Screenshot_20241005_194409_AnyDesk](https://github.com/user-attachments/assets/ef5079cf-83d7-4d66-9073-3a6a383d045e)

**Comment by pd5wl** [comment]
When i run ./update_cloudlog.sh i keep running in to a fastforward error.

>         application/language/french/gridsquares_lang.php
>         application/language/french/menu_
> Aborting
> Fast-forward failed; stopping here.

Any sugestions?

**Comment by magicbug** [comment]
My guess would be corrupt files, you can try "git stash" or you cantry manually downloading and replacing all the files

**Comment by pd5wl** [comment]
Full download works fine. 

Git stash gave the following: 

> git stash
> Saved working directory and index state WIP on master: 8e28c1d4 Merge pull request #2297 from int2001/adif_import_call
> 

But the wierd thing is that when i did a full copie of files (from the master zip) the next update it failed the ./update_cloudlog.sh

**Comment by pd5wl** [comment]
Did another full update. Lets see with the next version if git pull will work.

**Comment by Phoenixx4000** [comment]
chmod 755 cloudlog_update.sh
./cloudlog_update.sh

in case of fast forward error rename specified files and restart cloudlog_update.sh

in my case it worked.

**Reply by magicbug** [reply]
Pleased to hear it
